packages/dev/core/src/Materials/GreasedLine/greasedLinePluginMaterial.ts

```diff
@@ -289,8 +291,9 @@ export class GreasedLinePluginMaterial extends MaterialPluginBase implements IGr
         if (this._color) {
             uniformBuffer.updateColor3("grl_singleColor", this._color);
         }
-
-        uniformBuffer.setTexture("grl_colors", this.colorsTexture ?? GreasedLineMaterialDefaults.EmptyColorsTexture);
+        const texture = this.colorsTexture ?? GreasedLineMaterialDefaults.EmptyColorsTexture;
+        uniformBuffer.setTexture("grl_colors", texture);
+        uniformBuffer.updateFloat2("grl_textureSize", texture?.getSize()?.width ?? 1, texture?.getSize()?.height ?? 1);

```

Comment:
`texture` can't be null/empty, you should be able to write `texture.getSize().width`.

Reply:
I seem to get  - `error TS18047: 'texture' is possibly 'null'`, but if texture exists, `getSize()` and `width` are valid, should it rather be `texture?.getSize.width()` then?

Reply:
Yes, in this case `texture?.getSize().width ?? 1` is ok.